crates/typst/src/layout/abs.rs

```diff
@@ -148,7 +148,7 @@ impl Debug for Abs {
 
 impl Repr for Abs {
     fn repr(&self) -> EcoString {
-        repr::format_float(self.to_pt(), Some(2), "pt")
+        repr::format_float(self.to_pt(), Some(2), false, "pt")
```

Comment:
there are a bunch of these calls with `Some(2), false`. maybe we can also extract a semantic function for that like `display_float`. (which would then dispatch to format_float)

Reply:
Looking at the calls to `format_float`, there are only three different kinds of calls:
1. `(None, true, "")` for the `Repr`,
2. `(Some(2), false, "xx")` for values with units,
3. `(Some(3), true, "")` for components of colors.

I created a `format_float_with_unit` function for the second case and a `format_float_component` for the third one. This split seems kind of arbitrary, but I don't think this is really problematic.